hyper/http11/response.py

```diff
@@ -48,7 +49,8 @@ def __init__(self, code, reason, headers, sock, connection=None):
         # bother checking for content-length, we just keep reading until
         # we no longer can.
         self._expect_close = False
-        if b'close' in self.headers.get(b'connection', []):
+        if (b'close' in self.headers.get(b'connection', []) or
+                b'close' in self.headers.get(b'proxy-connection', [])):
             self._expect_close = True
```

Comment:
I think we cannot rely on a `proxy-connection: close` header being present. There is no specification requiring it (in fact, `proxy-connection` is not a specified header at all, it's just something that has evolved over time and has been argued as being a "misunderstanding" of how the `connection` header was supposed to work). I don't think there's anything wrong with saying `proxy-connection: close` on a `200 OK` to a `CONNECT`: in fact, if the response is HTTP/1.1 it might even be good practice to provide `Connection: close`, as the termination of the response body is nominally a connection termination.